### ISSUE TITLE
Unable to translate code with only 1 basic-block which jumps back to itself

### ISSUE THREAD

**Issue by adwait1-g** [revng/revng#85]
Consider the following code:

```asm
section .text                                                                   
    global _start                                                               
                                                                                
_start:                                                                         
    mov rax, 60                                                                 
    mov rdi, 1                                                                  
    inc rsi                                                                     
    inc r15                                                                     
    add r15, r14                                                                
    xor rbx, rbx                                                                
    jmp _start
```

I used nasm and ld to generate an executable. When tried to translate with rev.ng, got the following error:

```
revng/exp$ nasm code4.asm -f elf64
revng/exp$ ld code4.o -o code4
revng/exp$ ../orchestra/root/bin/revng translate ./code4
Assertion failed at /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/ProgramCounterHandler.cpp:690

**Targets.size() != 0**

 #0 0x00007fe3ed6ef2f9 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) /builds/gitlab/revng/orchestra/orchestra/sources/llvm/llvm/lib/Support/Unix/Signals.inc:568:13
 #1 0x00007fe3efc23fdb print_stack_trace() /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/Assert.cpp:20:1
 #2 0x00007fe3efc23f09 terminate() /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/Assert.cpp:24:3
 #3 0x00007fe3efc23ea0 report(char const*, char const*, unsigned int, char const*) /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/Assert.cpp:28:0
 #4 0x00007fe3efc30898 /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/ProgramCounterHandler.cpp:690:3
 #5 0x00000000004b094b llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>::~IRBuilder() /builds/gitlab/revng/orchestra/orchestra/root/include/llvm/IR/IRBuilder.h:2673:7
 #6 0x00000000004b094b ProgramCounterHandler::buildDispatcher(std::__1::vector<std::__1::pair<MetaAddress, llvm::BasicBlock*>, std::__1::allocator<std::__1::pair<MetaAddress, llvm::BasicBlock*> > >&, llvm::BasicBlock*, llvm::BasicBlock*, llvm::Optional<BlockType::Values>) const /builds/gitlab/revng/orchestra/orchestra/sources/revng/include/revng/Support/ProgramCounterHandler.h:206:3
 #7 0x00000000004aafb8 JumpTargetManager::rebuildDispatcher(std::__1::set<MetaAddress, std::__1::less<MetaAddress>, std::__1::allocator<MetaAddress> >*) /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:1126:37
 #8 0x00000000004aaad7 Logger<true>::isEnabled() const /builds/gitlab/revng/orchestra/orchestra/sources/revng/include/revng/Support/Debug.h:50:52
 #9 0x00000000004aaad7 JumpTargetManager::setCFGForm(CFGForm::Values, std::__1::set<MetaAddress, std::__1::less<MetaAddress>, std::__1::allocator<MetaAddress> >*) /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:1092:17
#10 0x00000000004acbc5 JumpTargetManager::harvestWithAVI() /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:0:5
#11 0x00000000004a995a JumpTargetManager::harvest() /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:1923:5
#12 0x00000000004a9138 std::__1::vector<std::__1::pair<MetaAddress, llvm::BasicBlock*>, std::__1::allocator<std::__1::pair<MetaAddress, llvm::BasicBlock*> > >::empty() const /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/vector:664:23
#13 0x00000000004a9138 JumpTargetManager::peek() /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:797:23
#14 0x0000000000462d34 std::__1::__tuple_leaf<0ul, MetaAddress&, false>& std::__1::__tuple_leaf<0ul, MetaAddress&, false>::operator=<MetaAddress>(MetaAddress&&) /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/tuple:265:22
#15 0x0000000000462d34 std::__1::enable_if<__tuple_assignable<std::__1::pair<MetaAddress, llvm::BasicBlock*>, std::__1::tuple<MetaAddress&, llvm::BasicBlock*&> >::value, std::__1::__tuple_impl<std::__1::__tuple_indices<0ul, 1ul>, MetaAddress&, llvm::BasicBlock*&>&>::type std::__1::__tuple_impl<std::__1::__tuple_indices<0ul, 1ul>, MetaAddress&, llvm::BasicBlock*&>::operator=<std::__1::pair<MetaAddress, llvm::BasicBlock*> >(std::__1::pair<MetaAddress, llvm::BasicBlock*>&&) /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/tuple:442:49
#16 0x0000000000462d34 std::__1::tuple<MetaAddress&, llvm::BasicBlock*&>& std::__1::tuple<MetaAddress&, llvm::BasicBlock*&>::operator=<std::__1::pair<MetaAddress, llvm::BasicBlock*>, void>(std::__1::pair<MetaAddress, llvm::BasicBlock*>&&) /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/tuple:948:21
#17 0x0000000000462d34 CodeGenerator::translate(llvm::Optional<unsigned long>) /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/CodeGenerator.cpp:1224:37
#18 0x00000000004d3017 main /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/Main.cpp:184:13
#19 0x00007fe3ecc70bf7 __libc_start_main /build/glibc-S9d2JN/glibc-2.27/csu/../csu/libc-start.c:344:0
#20 0x000000000041dab7 _start (/home/dell/Documents/projects/my_research/binary-analysis/revng/orchestra/root/bin/revng-lift+0x41dab7)
 #0 0x00007fe3ed6ef2f9 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) /builds/gitlab/revng/orchestra/orchestra/sources/llvm/llvm/lib/Support/Unix/Signals.inc:568:13
 #1 0x00007fe3ed6ed5a7 llvm::sys::RunSignalHandlers() /builds/gitlab/revng/orchestra/orchestra/sources/llvm/llvm/lib/Support/Signals.cpp:72:18
 #2 0x00007fe3ed6ef65c SignalHandler(int) /builds/gitlab/revng/orchestra/orchestra/sources/llvm/llvm/lib/Support/Unix/Signals.inc:410:1
 #3 0x00007fe3ed26a980 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x12980)
 #4 0x00007fe3ecc8dfb7 raise /build/glibc-S9d2JN/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #5 0x00007fe3ecc8f921 abort /build/glibc-S9d2JN/glibc-2.27/stdlib/abort.c:81:0
 #6 0x00007fe3efc23f0e /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/Assert.cpp:24:3
 #7 0x00007fe3efc23ea0 report(char const*, char const*, unsigned int, char const*) /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/Assert.cpp:28:0
 #8 0x00007fe3efc30898 /builds/gitlab/revng/orchestra/orchestra/sources/revng/lib/Support/ProgramCounterHandler.cpp:690:3
 #9 0x00000000004b094b llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>::~IRBuilder() /builds/gitlab/revng/orchestra/orchestra/root/include/llvm/IR/IRBuilder.h:2673:7
#10 0x00000000004b094b ProgramCounterHandler::buildDispatcher(std::__1::vector<std::__1::pair<MetaAddress, llvm::BasicBlock*>, std::__1::allocator<std::__1::pair<MetaAddress, llvm::BasicBlock*> > >&, llvm::BasicBlock*, llvm::BasicBlock*, llvm::Optional<BlockType::Values>) const /builds/gitlab/revng/orchestra/orchestra/sources/revng/include/revng/Support/ProgramCounterHandler.h:206:3
#11 0x00000000004aafb8 JumpTargetManager::rebuildDispatcher(std::__1::set<MetaAddress, std::__1::less<MetaAddress>, std::__1::allocator<MetaAddress> >*) /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:1126:37
#12 0x00000000004aaad7 Logger<true>::isEnabled() const /builds/gitlab/revng/orchestra/orchestra/sources/revng/include/revng/Support/Debug.h:50:52
#13 0x00000000004aaad7 JumpTargetManager::setCFGForm(CFGForm::Values, std::__1::set<MetaAddress, std::__1::less<MetaAddress>, std::__1::allocator<MetaAddress> >*) /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:1092:17
#14 0x00000000004acbc5 JumpTargetManager::harvestWithAVI() /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:0:5
#15 0x00000000004a995a JumpTargetManager::harvest() /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:1923:5
#16 0x00000000004a9138 std::__1::vector<std::__1::pair<MetaAddress, llvm::BasicBlock*>, std::__1::allocator<std::__1::pair<MetaAddress, llvm::BasicBlock*> > >::empty() const /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/vector:664:23
#17 0x00000000004a9138 JumpTargetManager::peek() /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/JumpTargetManager.cpp:797:23
#18 0x0000000000462d34 std::__1::__tuple_leaf<0ul, MetaAddress&, false>& std::__1::__tuple_leaf<0ul, MetaAddress&, false>::operator=<MetaAddress>(MetaAddress&&) /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/tuple:265:22
#19 0x0000000000462d34 std::__1::enable_if<__tuple_assignable<std::__1::pair<MetaAddress, llvm::BasicBlock*>, std::__1::tuple<MetaAddress&, llvm::BasicBlock*&> >::value, std::__1::__tuple_impl<std::__1::__tuple_indices<0ul, 1ul>, MetaAddress&, llvm::BasicBlock*&>&>::type std::__1::__tuple_impl<std::__1::__tuple_indices<0ul, 1ul>, MetaAddress&, llvm::BasicBlock*&>::operator=<std::__1::pair<MetaAddress, llvm::BasicBlock*> >(std::__1::pair<MetaAddress, llvm::BasicBlock*>&&) /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/tuple:442:49
#20 0x0000000000462d34 std::__1::tuple<MetaAddress&, llvm::BasicBlock*&>& std::__1::tuple<MetaAddress&, llvm::BasicBlock*&>::operator=<std::__1::pair<MetaAddress, llvm::BasicBlock*>, void>(std::__1::pair<MetaAddress, llvm::BasicBlock*>&&) /builds/gitlab/revng/orchestra/orchestra/root/bin/../include/c++/v1/tuple:948:21
#21 0x0000000000462d34 CodeGenerator::translate(llvm::Optional<unsigned long>) /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/CodeGenerator.cpp:1224:37
#22 0x00000000004d3017 main /builds/gitlab/revng/orchestra/orchestra/sources/revng/tools/revng-lift/Main.cpp:184:13
#23 0x00007fe3ecc70bf7 __libc_start_main /build/glibc-S9d2JN/glibc-2.27/csu/../csu/libc-start.c:344:0
#24 0x000000000041dab7 _start (/home/dell/Documents/projects/my_research/binary-analysis/revng/orchestra/root/bin/revng-lift+0x41dab7)
The following command exited with -6:
/home/dell/Documents/projects/my_research/binary-analysis/revng/orchestra/root/bin/revng-lift -g ll code4 code4.translated.ll
```

There is one jump-target - the basic-block itself. But it is not considered. I started trying with the following:

```asm
_start:
       jmp _start
```

Got the same error.

Thanks,
Adwaith

**Comment by aleclearmind** [comment]
Thanks for reporting this issue, we'll look into this ASAP.

**Comment by aleclearmind** [comment]
Sorry for the long silence.
I can now successfully translate your example program. An unrelated bug has emerged while testing this and it was fixed in dd5914c.
Thanks for reporting this, feel free to reopen in case you still have issues.
